packages/central-server/src/apiV2/surveyResponse.js

```diff
@@ -118,7 +118,14 @@ export const submitResponses = async (models, userId, responses) => {
 };
 
 export async function surveyResponse(req, res) {
-  const { userId, body, models } = req;
+  const { userId, body, query, models } = req;
+  const { submitAsPublic } = query;
```

Comment:
What are your thoughts on just always using the public user if `userId` is not set? Would save having to do any checks in the frontend and the orchestration server

Reply:
The userId is actually always set. It's not very intuitive but what actually happens is when there is no real user logged in, there is a special api user that makes the requests between servers and so the userId will be that one here is there is no on logged in.